src/SFML/Window/Unix/Display.cpp

```diff
@@ -73,31 +67,20 @@ Display* openDisplay()
         }
     }
 
-    ++referenceCount;
     return sharedDisplay;
 }
 
 
 ////////////////////////////////////////////////////////////
-void closeDisplay(Display* display)
-{
-    const std::lock_guard lock(mutex);
-
-    assert(display == sharedDisplay && "Display must match shared display");
-
-    --referenceCount;
-    if (referenceCount == 0)
-        XCloseDisplay(display);
-}
-
-////////////////////////////////////////////////////////////
-XIM openXim()
+std::shared_ptr<_XIM> openXim()
 {
     const std::lock_guard lock(mutex);
 
     assert(sharedDisplay != nullptr && "Shared display is null. Call priv::openDisplay() to initialize it.");
 
-    if (referenceCountXIM == 0)
```

Comment:
I'm not sure why we need this `assert`. It seems more reasonable to call `openDisplay` inside this function to ensure a display is open rather than hope this precondition is always true. That seemed out of scope for this PR so I didn't make that change.